codeserver/ubi9-python-3.12/run-nginx.sh

```
@@ -6,28 +6,87 @@ set -e
 else
     export BASE_URL=$(echo $NB_PREFIX | awk -F/ '{ print $4"-"$3 }')$(echo $NOTEBOOK_ARGS | grep -Po 'hub_host":"\K.*?(?=")' | awk -F/ '{ print $3 }' | awk -F. '{for (i=2; i<=NF; i++) printf ".%s", $i}')
     envsubst '${NB_PREFIX},${BASE_URL}' < /opt/app-root/etc/nginx.default.d/proxy.conf.template_nbprefix > /opt/app-root/etc/nginx.default.d/proxy.conf
 
-    # A temp file is used because piping envsubst directly into the same file via `tee`
-    # is a race condition: `tee` can truncate the file before `envsubst` finishes reading it,
-    # resulting in an empty/corrupt config and "no events section" errors from nginx.
-    tmp=$(mktemp)
-    envsubst '${BASE_URL}' < /etc/nginx/nginx.conf > "$tmp"
-    cat "$tmp" > /etc/nginx/nginx.conf
-    rm -f "$tmp"
+    tmp=\$(mktemp)
```

Shellcheck (0.11.0)

[error] '(' is invalid here. Did you forget to escape it?

(SC1036)